docker-images/Dockerfile

```diff
@@ -0,0 +1,23 @@
+ARG PYTHON_VERSION=3.11.7
+FROM python:${PYTHON_VERSION}-slim as base
+
+# Prevents Python from writing pyc files.
+ENV PYTHONDONTWRITEBYTECODE=1
+
+# Keeps Python from buffering stdout and stderr to avoid situations where
+# the application crashes without emitting any logs due to buffering.
+ENV PYTHONUNBUFFERED=1
+
+WORKDIR .. .
+
+# Download dependencies from pyproject.toml
+#RUN pip install
+
+# Copy the source code into the container.
+COPY .. .
+
+# Expose the port that the application listens on.
+EXPOSE 8000
+
+# Run the application.
+CMD poetry run gunicorn -w 5 -b 0.0.0.0:8000 page_analyzer:app
```

Comment:
`CMD` written as list of strings (aka "exec form") is the best practice - https://docs.docker.com/reference/dockerfile/#cmd